### DISCUSSION TITLE
Allow Custom CSS Classes on All Block Elements?

### DISCUSSION THREAD

**Discussion by shustudios** [ueberdosis/tiptap#1792] [General Questions] [answered]
Hi, I would like to give my users the ability to add custom css classes to any of the block level elements in the editor. I've been able to do this with the Paragraph node by creating a custom Paragraph extension, however, this seems a bit overkill to have to do it for every node type I wish to target.

Is there a global way to do this?

Sidenote - I would also like to be able to wrap any selected text in a <span> tag with a custom class, but I figure I should start off simple and just focus on the block level elements first :)

**Comment by philippkuehn** [comment, marked as answer]
you can add classes like this: https://www.tiptap.dev/guide/styling/#option-2-add-custom-classes

**Reply by shustudios** [reply]
Sorry - I meant adding custom classes **while editing** - for example, the same way you could highlight text and make a node a heading, but instead, add/remove a (user defined) custom css class.

**Reply by philippkuehn** [reply]
Then you should register a global class attribute: https://www.tiptap.dev/guide/custom-extensions#global-attributes

**Reply by shustudios** [reply]
Ah - perfect! I'll look into this. Thanks!
